lib/streamlit/commands/execution_control.py

```diff
@@ -87,3 +90,49 @@ def experimental_rerun() -> NoReturn:
     # be seen.
     _LOGGER.warning(msg)
     rerun()
+
+
+@gather_metrics("switch_page")
+def switch_page(page: str) -> NoReturn:
+    """Switch the current programmatically page in a multi-page app.
+    When `st.switch_page()` is called with a page, the current page script is halted
+    and the requested page script will be queued to run from the top.
+    Parameters
+    ----------
+    page: str
+        The file path, relative to the main script, of the page to switch to.
+    """
+
+    ctx = get_script_run_ctx()
+
+    query_string = ""
+    ctx_main_script = ""
+    if ctx:
+        query_string = ctx.query_string
+        ctx_main_script = ctx.main_script_path
+
+    main_script_path = os.path.join(os.getcwd(), ctx_main_script)
+    main_script_directory = os.path.dirname(main_script_path)
+
+    if page.startswith("./"):
+        page = page[2:]
+    elif page.startswith("/"):
+        page = page[1:]
+
+    requested_page = os.path.join(main_script_directory, page)
+    all_app_pages = source_util.get_pages(ctx_main_script).values()
+
+    for page_data in all_app_pages:
+        full_path = page_data["script_path"]
+
+        if requested_page == full_path:
+            raise RerunException(
```

Comment:
I think we will want to do something like.

```python
ctx.script_requests.request_rerun(
    RerunData(
        query_string=query_string, # maybe not query_string
        page_script_hash=page_script_hash,
     )
 )
```

See https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/commands/execution_control.py#L53

Reply:
I would think we should have to do the same thing here.